openassessment/xblock/workflow_mixin.py

```diff
@@ -126,13 +126,47 @@ def get_workflow_info(self, submission_uuid=None):
             AssessmentWorkflowError
         """
         if submission_uuid is None:
-            submission_uuid = self.submission_uuid
-            if submission_uuid is None:
-                return {}
+            submission_uuid = self.try_get_submission_uuid()
+
+        if submission_uuid is None:
+            return {}
         return workflow_api.get_workflow_for_submission(
             submission_uuid, self.workflow_requirements()
         )
 
+    def try_get_submission_uuid(self):
```

Comment:
Just `get_submission_uuid()` is fine - it's pretty common to have methods that return either "the thing" or None, and we call them `get_the_thing()`.